lib/auth/auth.go

```diff
@@ -554,6 +568,7 @@ func (a *Server) updateVersionMetrics() {
 
 func (a *Server) Close() error {
 	a.cancelFunc()
+	a.inventory.Close()
```

Comment:
I don't feel too strongly either way about the `inventory.Controller` in particular, but I've come around to the perspective that explicit close is better for persistent (i.e. not request scoped) resources/components.  Not sure I'm *right* about that, but here are some of my thoughts on it (in no particular order):

- Its nice to be able to control the order in things close and/or be able to easily inject things that you might want to happen "before" a specific component starts to close.  In particular, it is much easier to introduce a "graceful shutdown" mode later down the line if closure is compartmentalized from the start, rather than closure basically happening at every level all at once.
- Its generally easier to switch from explicit close to implicit close than it is to switch from implicit close to explicit close, so if you're not sure which one you should use, seems best to default to explicit.
- In the broader go ecosystem contexts are almost never used for the closure of long-lived resources.  Not necessarily a good reason not to do it, but I can think of at least two reason for why explicit closure seems to be preferred by most folks over implicit closure:
  1. Implicit closure means that you might never actually get to do what you want to do as part of your particular closure process.  Once main exits, all background goroutines terminate without even being able to run deferred functions.  Any component that relies on context cancellation to begin its cleanup process may simply never have its cleanup run.  This is fine for a lot of components, but implicit closure is contagious.  If any component "above" you is implicitly closed, then you are in effect also implicitly closed.  This isn't a disaster... programs exit unexpectedly sometimes and should be able to handle that, but it still seems that we aught to try to respect Close() methods where we can.
  2. Implicit/asynchronous closure is incompatible with finite resources. E.g. if a component opens a listener on some port P, its reasonable to assume that two of said component can't be run at the same time, but one aught to feel safe in the assumption that once Close() has been called on the first instance, starting a second should now succeed. If all cleanup is asynchronous (and, as mentioned above, implicit/asynchronous closure is contagious) then that can't be relied on.
  
Like I said... I don't think it really matters in this case, but for the above reasons I've been trying to default to explicit close unless I can think of some specific reason why implicit close is better in the particular case I'm looking at.